server/hotrod/src/main/java/org/infinispan/server/hotrod/Decoder2x.java

```diff
@@ -411,6 +411,13 @@ public void customReadKey(HotRodHeader header, ByteBuf buffer, CacheDecodeContex
             }).isPresent()) {
                return;
             }
+            if (Constants.isVersionPost25(header.version)) {
+               int listenerInterests = ExtendedByteBufJava.readMaybeVInt(buffer);
+               if (listenerInterests == Integer.MIN_VALUE)
+                  return;
+
+               requestCtx.setListenerInterests(listenerInterests);
+            }
```

Comment:
Hmmmmm, not sure exactly how you expect the code to look. It's quite confusing the way it currently looks. Can you paste the final code somewhere or send me a PR?